dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+            # with respect to the randomness choice of pairs of edges considered in the inner loop
+            adj_graph.remove_edges_from(self.remove_edges)
+
+            # increment the conditioning set size
+            size_cond_set += 1
+
+            # only allow conditioning set sizes up to maximum set number
+            if size_cond_set > self.max_cond_set_size_ or cont is False:
+                break
+
+        self.adj_graph_ = adj_graph
+
+    def _summarize_xy_comparison(self, x_var, y_var, removed_edge, pvalue):
+        # exit loop if we have found an independency and removed the edge
+        if removed_edge:
+            remove_edge_str = "Removing edge"
+        else:
+            remove_edge_str = "Did not remove edge"
+
+        logger.info(
+            f"{remove_edge_str} between {x_var} and {y_var}... \n"
+            f"Statistical summary:\n"
+            f"- PValue={pvalue} at alpha={self.alpha}"
+        )
+
+    def _compute_candidate_conditioning_sets(
+        self, adj_graph: nx.Graph, x_var, y_var, skeleton_method: str
+    ) -> Set[Any]:
+        """Compute candidate conditioning sets.
+
+        Parameters
+        ----------
+        adj_graph : nx.Graph
+            The current adjacency graph.
+        x_var : node
+            The 'X' node.
+        y_var : node
+            The 'Y' node.
+        skeleton_method : str
+            The skeleton method, which dictates how we choose the corresponding
+            conditioning sets.
+
+        Returns
+        -------
+        possible_variables : Set
+            The set of nodes in 'adj_graph' that are candidates for the
+            conditioning set.
+        """
+        if skeleton_method == "complete":
+            possible_variables = set(adj_graph.nodes)
+        elif skeleton_method == "neighbors":
+            possible_variables = set(adj_graph.neighbors(x_var))
+            # possible_adjacencies.copy()
+        elif skeleton_method == "neighbors_path":
+            # constrain adjacency set to ones with a path from x_var to y_var
+            possible_variables = _find_neighbors_along_path(adj_graph, start=x_var, end=y_var)
+
+        if self.keep_sorted:
+            possible_variables = sorted(
+                possible_variables,
+                key=lambda n: adj_graph.edges[x_var, n]["test_stat_abs"],
+                reverse=True,
+            )  # type: ignore
+
+        if x_var in possible_variables:
+            possible_variables.remove(x_var)
+        if y_var in possible_variables:
+            possible_variables.remove(y_var)
+
+        return possible_variables
+
+    def _postprocess_ci_test(
+        self, adj_graph: nx.Graph, x_var, y_var, cond_set: Set, test_stat: float, pvalue: float
+    ) -> bool:
+        # keep track of the smallest test statistic, meaning the highest pvalue
+        # meaning the "most" independent
+        if np.abs(test_stat) <= adj_graph.edges[x_var, y_var]["test_stat_abs"]:
```

Comment:
Wouldn't it be enough to store ```test_stat``` instead of ```test_stat_abs```? If the absolute value is needed, one can simply add an ```abs(..)```.

Reply:
Sure that makes sense.